# Invasive Escherichia coli in neonates in Chatinkha nursery

Supplementary materials and code generating figures for a paper describing the genomics of invasive E. coli in neonates in Malawi. Here we use whole genome sequencing to describe the antigenic and overall diversity of Escherichia coli isolates causing invasive infections in neonates in a single hospital in Malawi. We show high levels of O-antigen diversity, highlighting the difficulty of developing a vaccine in this patient population and emphasizing the need for more data from this patient population.

## Supplementary data

Supplementary data is in the data folder, and includes five files:

1) Accession numbers for short read data: "Supp_data_1_shortreads.csv"
2) Accession numbers for long read data: "Supp_data_2_longreads.csv"
3)  Overall blood culture numbers for QECH: "Supp_data_4_culture.csv"
4) Individual patient information and genomic data: "Supp_data_3_genomes.csv"
5) AMR genes, outputs from AMRfinder: "Supp_data_5_amr.csv"

## Figures and Supplementary figures

The Rmarkdown document ecoli_invasive_markdown.Rmd contains the code to generate all of the figures apart from Supplementary Figure 2. This is in the file Supplementary Figure 2. You will require R, Rstudio and to install several R packages to run this successfully. 

## Citation

If you use the data from this repository please cite:

**Add paper citation when it is published**

High diversity of Escherichia coli causing invasive disease in neonates in Malawi poses challenges for O-antigen based vaccine approach
Oliver Pearse, Allan Zuza, Edith Tewesa, Patricia Siyabuh, Alice J Fraser, Jennifer Cornick, Kondwani Kawaza, Patrick Musicha, Nicholas R Thomson, Nicholas A Feasey, Eva Heinz
medRxiv 2024.09.24.24314145; doi: https://doi.org/10.1101/2024.09.24.24314145

## License

This project is licensed under the Creative Commons Attribution 4.0 International License (CC BY 4.0).  
See the licence.txt file for details.
